Library/Formula/asio.rb

```diff
@@ -0,0 +1,61 @@
+class Asio < Formula
+  desc "Cross-platform C++ Library for asynchronous programming"
+  homepage "http://think-async.com/Asio"
+  url "https://downloads.sourceforge.net/project/asio/asio/1.10.6%20%28Stable%29/asio-1.10.6.tar.bz2"
+  sha256 "e0d71c40a7b1f6c1334008fb279e7361b32a063e020efd21e40d9d8ff037195e"
+  head "https://github.com/chriskohlhoff/asio.git"
+
+  devel do
+    url "https://downloads.sourceforge.net/project/asio/asio/1.11.0%20%28Development%29/asio-1.11.0.tar.bz2"
+    sha256 "4f7e13260eea67412202638ec111cb5014f44bdebe96103279c60236874daa50"
+  end
+
+  option "with-boost-coroutine", "use Boost.Coroutine to implement stackful coroutines"
+  option :cxx11
+  needs :cxx11 if build.cxx11?
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "boost" => :optional
+  depends_on "boost" unless build.cxx11?
+  depends_on "boost" if build.with? "boost-coroutine"
+  depends_on "openssl"
+
+  def install
+    if build.head?
+      cd "asio"
+      system "./autogen.sh"
+    else
+      system "autoconf"
+    end
+    ENV.cxx11 if build.cxx11?
+    args = %W[
+      --disable-dependency-tracking
+      --disable-silent-rules
+      --prefix=#{prefix}
+      --with-boost=#{(build.with?("boost") || build.with?("boost-coroutine") || !build.cxx11?) ? Formula["boost"].opt_include : "no"}
+    ]
+    args << "--enable-boost-coroutine" if build.with? "boost-coroutine"
+    system "./configure", *args
+    system "make", "install"
+    pkgshare.install "src/examples"
+  end
+
+  test do
+    if build.cxx11?
```

Comment:
We only support testing default options. We're not prepared to open the vast floodgates to testing non-default options. The primary purpose of `test do` is to serve the test-bot and ensure the formula functions as expected both now & long-term.